examples/miral-custom-decorations/user_decoration_manager_example.cpp

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "input.h"
+#include "user_decoration_example.h"
+#include "user_decoration_manager_example.h"
+#include "miral/decoration/decoration_adapter.h"
+
+#include <mir/shell/decoration.h>
+#include <mir/graphics/display_configuration.h>
+#include <mir/graphics/null_display_configuration_observer.h>
+/* #include <bffoost/throw_exception.hpp> */
+#include "mirserver-internal/mir/observer_registrar.h"
```

Comment:
Is this okay, or should observer_registrar.h be exposed / I should access it some other way?